tests/debugging/test_encoding.py

```diff
@@ -265,3 +266,27 @@ def test_format_message(args, expected):
 
 def test_encoding_none():
     assert utils.capture_value(None) == {"isNull": True, "type": "NoneType"}
+
+
+def test_encoding_timeout():
```

Comment:
While I love the idea here, I think this can be flaky. I suggest create a mocked version hg that return false after X invocations to avoid time flakiness

Reply:
I think that with steps by an order of magnitude we should be fine here. If this becomes an issue we can refactor it. How does that sound?

Reply:
I get the point of see if this flaky and fix. but the more I think about it - it doesn't test the functionality of the class.

I think we need to have tests to validate:
1. we stop capture in middle of a list.
2. we stop capture in a middle of a dict
3. we stop capture in the middle of an object (fields).

combined with simplified interface for captured value (cancellation function) I think we can change this test to validate those condition specifically and testing random exit point using time.